### PR TITLE
Publish pipeline options to flink web ui

### PR DESCRIPTION
**Pull request by kkdoon**
This PR adds functionality to publish the job's pipeline option values to Flink's web UI (issue #26822). This is quite useful for debugging purposes where users can inspect the runtime arguments passed to the beam job, for flink runner.

image
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by kkdoon** [comment]
R: @mxm @damccorm

**Comment by kkdoon** [comment]
Hi @mxm, would you be able to review this PR or point me to someone who would be the best candidate for this review?

Thanks!

**Comment by kkdoon** [comment]
Hi @damccorm, pinging you for review of this PR. Tagged you since you were the last one to update this file. Let me know if someone else would be better suited to review this PR. Thanks!

**Comment by damccorm** [comment]
Hey @kkdoon unfortunately that was a non-functional change and I don't really have the context needed here.

@je-ik or @xinyuiscool would either of you be good reviewers for this change?

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java [hunk 1]
```diff
@@ -140,6 +150,9 @@ static ExecutionEnvironment createBatchExecutionEnvironment(
 
     applyLatencyTrackingInterval(flinkBatchEnv.getConfig(), options);
 
+    configureWebUIOptions(
+        flinkBatchEnv.getConfig(), options.as(org.apache.beam.sdk.options.PipelineOptions.class));
```
[the remark this answers is not in the document]
Comment (kkdoon) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java:154]: good catch! will import it instead.

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java [hunk 2]
```diff
@@ -245,9 +258,63 @@ public static StreamExecutionEnvironment createStreamExecutionEnvironment(
...
+
+    try {
+      JsonNode node = mapper.readTree(optionsAsString);
+      JsonNode optionsNode = node.get("options");
+      Map<String, String> output =
+          Streams.stream(optionsNode.fields())
+              .filter(
+                  entry ->
+                      !entry.getValue().asText().isEmpty()
+                          && !"null".equals(entry.getValue().asText()))
```
[the remark this answers is not in the document]
Comment (kkdoon) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java:281]: good point, i can use `isNull` method in JsonNode class to detect null scenarios.

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java [hunk 3]
```diff
@@ -245,9 +258,63 @@ public static StreamExecutionEnvironment createStreamExecutionEnvironment(
...
+    String optionsAsString = serializablePipelineOptions.toString();
+
+    try {
+      JsonNode node = mapper.readTree(optionsAsString);
+      JsonNode optionsNode = node.get("options");
+      Map<String, String> output =
+          Streams.stream(optionsNode.fields())
+              .filter(
+                  entry ->
+                      !entry.getValue().asText().isEmpty()
```
[the remark this answers is not in the document]
Comment (kkdoon) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java:280]: makes sense, i will remove this check. `IsNull` is sufficient to check whether the field is set or not.

**Comment by kkdoon** [comment]
Hi @mxm, I addressed your review comments. Could you kindly re-review it again, when you get a chance. 

Thanks!

**Comment by kkdoon** [comment]
@mxm re-pinging for review. Thanks!

**Review by mxm — APPROVED** [review]
Thanks @kkdoon!

**Comment by mxm** [comment]
Sorry for the delay on my side!

**Comment by kkdoon** [comment]
Thanks @mxm for the approval! Could you merge it from your side or something is pending from my side first?

**Comment by kkdoon** [comment]
Run Java PreCommit

**Comment by mxm** [comment]
@kkdoon Could you please file a JIRA for this change or link an existing one? Apologies, I just saw that the change doesn't have a JIRA associated yet.

**Comment by kkdoon** [comment]
@mxm i created issue #26822

**Comment by kkdoon** [comment]
@mxm re-pinging to see if it can be merged now. Thanks!

**Comment by mxm** [comment]
Thanks @kkdoon!!

**Pull request merged**